bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/dataexchange/NukiApiServlet.java

```diff
@@ -137,33 +139,70 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
                         ? (String) thing.getConfiguration().get(NukiBindingConstants.CONFIG_NUKI_ID)
                         : null;
                 if (nukiIdThing != null && nukiIdThing.equals(nukiId)) {
-                    logger.debug("Processing ThingUID[{}] - nukiId[{}]", thing.getUID(), nukiId);
-                    NukiSmartLockHandler nsh = getSmartLockHandler(thing);
-                    if (nsh == null) {
-                        logger.debug("Could not update channels for ThingUID[{}] because Handler is null!",
-                                thing.getUID());
+                    logger.debug("Processing ThingUID[{}] - nukiId[{}] - Thing Type UID[{}]", thing.getUID(), nukiId,
+                            thing.getThingTypeUID());
+                    if (thing.getThingTypeUID().equals(NukiBindingConstants.THING_TYPE_SMARTLOCK)) {
+                        NukiSmartLockHandler nsh = getSmartLockHandler(thing);
+                        if (nsh == null) {
+                            logger.debug("Could not update channels for ThingUID[{}] because Handler is null!",
+                                    thing.getUID());
+                            break;
+                        }
+                        Channel channel = thing.getChannel(NukiBindingConstants.CHANNEL_SMARTLOCK_LOCK);
+                        if (channel != null) {
+                            State state = bridgeApiLockStateRequestDto
+                                    .getState() == NukiBindingConstants.LOCK_STATES_LOCKED ? OnOffType.ON
+                                            : OnOffType.OFF;
+                            nsh.handleApiServletUpdate(channel.getUID(), state);
+                        }
+                        channel = thing.getChannel(NukiBindingConstants.CHANNEL_SMARTLOCK_STATE);
+                        if (channel != null) {
+                            State state = new DecimalType(bridgeApiLockStateRequestDto.getState());
+                            nsh.handleApiServletUpdate(channel.getUID(), state);
+                        }
+                        channel = thing.getChannel(NukiBindingConstants.CHANNEL_SMARTLOCK_LOW_BATTERY);
+                        if (channel != null) {
+                            State state = bridgeApiLockStateRequestDto.isBatteryCritical() ? OnOffType.ON
+                                    : OnOffType.OFF;
+                            nsh.handleApiServletUpdate(channel.getUID(), state);
+                        }
+                        setHeaders(response);
+                        response.getWriter().println(gson.toJson(new NukiHttpServerStatusResponseDto("OK")));
+                        return;
+                    } else if (thing.getThingTypeUID().equals(NukiBindingConstants.THING_TYPE_OPENER)) {
+                        NukiOpenerHandler nsh = getOpenerHandler(thing);
```

Comment:
Might be out of scope of this PR but it would be much safer to check the handler instance type through `instanceof` rather than checking against the thingTypeUID.